### PR TITLE
Update to 1.0.0

### PR DESCRIPTION
**Pull request by thehappydinoa**
- Implemented `typing`
- Implemented `mypy`
- Implemented `flake8`
- Implemented `black`
- Implemented `pytest`
- Implemented `pytest-cov`
- Fixed timeout implementation
- Removed `requirements.txt`
- Removed unused scripts
- Bumped `__version__`
- Updated `setup.py`
- Updated `.gitignore`
- Updated `README.md`
- Added `CONTRIBUTING.md`
- Added `MANIFEST.in`
- Added [docstrings](https://www.python.org/dev/peps/pep-0257/)
- Added `config.py`
- Added `cli.py` from [censys/censys-command-line](https://github.com/censys/censys-command-line)
- Added `hnri` subcommand
- Added `config` subcommand
- Moved `search` to a subcommand
- Added `sphinx` docs
- Added `.readthedocs.yml`

### CODE REVIEW

File: censys/__init__.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 __author__ = "Censys Team"
 __email__ = "support@censys.io"
-__copyright__ = "Copyright 2017 ZCorp Technology"
-__version__ = "0.1.0"
+__copyright__ = "Copyright 2020 Censys, Inc."
```
Comment (bwireman) [inline censys/__init__.py:3]: idk enough about anything to know if this is ok

File: README.md [hunk 2]
```diff
@@ -1,105 +1,168 @@
...
+
+## Developing
+
+Install dev dependencies with:
+
+```bash
+pip install -e .[dev]
+```
 
-series_name = '22-ssh-banner-full_ipv4'
```
Comment (bwireman) [inline README.md:97]: This code still works, we just aren't producing new scan specific datasets. Not sure if we should keep it

Comment (thehappydinoa) [reply README.md:97]: I believe this format for series name was replaces by one including date.

File: censys/__init__.py (hunk 1 above)
Comment (thehappydinoa) [reply censys/__init__.py:3]: I believe that the copyright is just displayed in PyPi.

File: README.md [hunk 3]
```diff
@@ -1,105 +1,168 @@
...
 
-Usage
-=====
+```bash
+pip install censys
+```
+
+## Usage
 
 There are three API endpoints that the library provides access to: Index,
```
Comment (thehappydinoa) [inline README.md:15]: @bwireman Here we reference Index, Query, and Export. Do we still use these terms?

File: README.md (hunk 2 above)
Comment (thehappydinoa) [reply README.md:97]: Added in [`README.md#107`](https://github.com/censys/censys-python/blob/eaff32d828bfc606a713293964858b6836dffae6/README.md#data)

File: censys/cli.py [hunk 4]
```diff
@@ -0,0 +1,685 @@
...
+        This method writes the search results to screen.
+
+        Args:
+            search_results (Results): list of search results from API query.
+
+        Returns:
+            bool: True if wrote to file successfully.
+        """
+
+        print(json.dumps(search_results, indent=4))
```
Comment (thomas-censysio) [inline censys/cli.py:117]: Was going to suggest a JSON output for dumping to screen (ex: pipe through `jq`) but it looks like that's what "screen" already does :)

File: censys/cli.py [hunk 5]
```diff
@@ -0,0 +1,685 @@
...
+        index = CensysWebsites(api_id=self.api_user, api_secret=self.api_pass)
+
+        return self._process_search(
+            query,
+            index,
+            self._combine_fields(default_fields, fields, overwrite=overwrite),
+        )
+
+
+class CensysHNRI:
```
Comment (thomas-censysio) [inline censys/cli.py:336]: Is this exactly what HNRI does for risk computation? If not - it might create some confusion on the HNRI product.

**Review by thomas-censysio — APPROVED** [review]
Looks mostly good. The usage of the term HNRI makes me a little hesitant as I understand the intended purpose, however it may lead to product confusion based on what our paid HNRI product does.

File: censys/cli.py (hunk 5 above)
Comment (thehappydinoa) [reply censys/cli.py:336]: It uses the exact same code from me.censys.io. Should I rename it to `me`?

File: censys/cli.py (hunk 4 above)
Comment (thehappydinoa) [reply censys/cli.py:117]: Ah ya, I have been debating between `json.dumps` and `pprint`. Any thoughts?

Comment (thomas-censysio) [reply censys/cli.py:117]: `pprint` might be more appropriate for 'screen' output. Having an option to optionally indicate target file - perhaps w/ `-` indicating output to stdout might be useful to facilitate chaining tools.

Comment (thehappydinoa) [reply censys/cli.py:117]: Okay I will make `--format` and `--output` as well as switch to `pprint`

Comment (thehappydinoa) [reply censys/cli.py:117]: Resolved in 1a182cc

File: censys/base.py [hunk 6]
```diff
@@ -1,176 +1,294 @@
...
+class CensysAPIBase:
+    """
+    This class is the base for API queries.
+
+    Args:
+        api_id (str): The API id provided by Censys.
+        api_secret (str): The API secret provided by Censys.
+        url (str): Adjust the number of results returned by the API.
+        timeout (int): Timeout for API requests.
+        user_agent_identifier (str): User agent string to send.
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply censys/base.py]: I based it on https://requests.readthedocs.io/en/master/community/faq/#custom-user-agents, but I can add some language from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

File: censys/cli.py (hunk 5 above)
Comment (thomas-censysio) [reply censys/cli.py:346]: After some outside discussion, HNRI does make sense here (it's what we call it on me.censys.io) - the "enhanced" version we offer through the SAS product can be seen as sufficiently distinct as the tools offered here are more "DIY".

Comment (thehappydinoa) [reply censys/cli.py:346]: Sounds good 👍

**Review by mehdimas — COMMENTED** [review]
Just some minor things.

**Review by mehdimas — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,105 +1,249 @@
...
+# Get bulk certificate data
+certs = c.bulk(fingerprints)
+print(certs)
+```
+
+## CLI Usage
+
+### Search
+
+```bash
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply README.md]: I actually moved the CLI to be auto generated by Sphinx.

File: README.md [hunk 8]
```diff
@@ -1,105 +1,249 @@
...
+
+Below we show an example using the `CensysIPv4` index.
+
+```python
+import censys.ipv4
+
+c = censys.ipv4.CensysIPv4(api_id="XXX", api_secret="XXX")
+
+for page in c.search(
+    "443.https.get.headers.server: Apache AND location.country: Japan", max_records=10
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply README.md]: Fixed in 5bf8b7e

File: README.md [hunk 9]
```diff
@@ -1,105 +1,249 @@
...
-		 '110.pop3.starttls.tls.certificate.parsed.subject_dn']
-
-data = list(c.search("80.http.get.headers.server: Apache AND location.country: Japan", 
-                             IPV4_FIELDS, max_records=10))		 
-print (data)
+# Gets account data
+c.account()
+
+# Gets account quota
+c.quota()
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply README.md]: Fixed in 5bf8b7e

**Comment by andy-censys** [comment]
@zakird and @mattclement this would be useful to have available if you have time to give it a glance.  Thanks, --andy

**Comment by zakird** [comment]
Please don't block on me for this one. If the dev team is happy with the code, let's merge!

File: censys/config.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+import os
+from pathlib import Path
+from configparser import ConfigParser, NoOptionError
+
+from censys import __version__
+
+DEFAULT = "DEFAULT"
+
+censys_path = os.path.join(str(Path.home()), ".censys")
```
Comment (mattclement) [inline censys/config.py:13]: While this isn't _exactly_ the XDG spec, `.config` is where this stuff should go.
```suggestion
censys_path = os.path.join(str(Path.home()), ".config/censys")
```

File: censys/base.py [hunk 11]
```diff
@@ -1,176 +1,294 @@
...
+        url (str, optional): The URL to make API requests.
+        timeout (int, optional): Timeout for API requests in seconds.
+        user_agent_identifier (str, optional): Override User-Agent string.
+
+    Raises:
+        CensysAPIException: Base Exception Class for the Censys API.
+    """
+
+    DEFAULT_URL: str = "https://censys.io/api/v1"
+    """Default API base URL."""
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply censys/base.py:42]: It is, I decided on using the Google syntax for docstrings.

File: README.md [hunk 12]
```diff
@@ -1,105 +1,47 @@
...
-# iterate over certificates that match a search
-fields = ["parsed.subject_dn", "parsed.fingerprint_sha256"]
-for cert in c.search("github.com and valid_nss: true", fields=fields):
-	fingerprints.append(cert["parsed.fingerprint_sha256"])
-	print (cert["parsed.subject_dn"])
+## Contributing
 
-# get certificates in bulk from a list of fingerprints
-print (c.bulk(fingerprints))
+All contributions (no matter if small) are always welcome.
```
[the remark this answers is not in the document]
Comment (thehappydinoa) [reply README.md]: Fixed in 523b930

File: censys/config.py (hunk 10 above)
Comment (thehappydinoa) [reply censys/config.py]: Added in 7a0073f

**Pull request merged**